### PR TITLE
LT02 203 and LT02 204

### PR DESCRIPTION
**Pull request by lcarrietas**
Machine Learning module re-formed to app schema. 
Django and Postgresql docker image communication.
Checksum validation.
Basic exceptions handling (enhancement soon!)

### CODE REVIEW

File: MACHINE_LEARNING/image_recognizer/image_recognizer/asgi.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+"""
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer/asgi.py:12]: Please add the copyright.

File: MACHINE_LEARNING/image_recognizer/image_recognizer/urls.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+"""image_recognizer URL Configuration
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer/urls.py:12]: Please add the copyright.

File: MACHINE_LEARNING/image_recognizer/image_recognizer/wsgi.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+"""
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer/wsgi.py:12]: Please add the copyright.

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/admin.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/admin.py:12]: Please add the copyright.

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/apps.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+from django.apps import AppConfig
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/apps.py:12]: Please add the copyright.

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/exceptions/file_exception.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class FileException(Exception):
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/exceptions/file_exception.py:12]: Please add the copyright.

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/exceptions/zip_exception.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class ZipException(Exception):
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/exceptions/zip_exception.py:12]: Please add the copyright.

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/model/prediction.py [hunk 8]
```diff
@@ -16,6 +16,8 @@
 from .result import Result
 from .vgg16 import Vgg16
 from .yolo import Yolo
+#from .exceptions.file_exception import FileException
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/model/prediction.py]: Please check the duplicate import

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/models.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    id = models.AutoField(primary_key=True)
+    name = models.CharField(max_length=50)
+    path = models.CharField(max_length=50)
+    checksum = models.CharField(max_length=100)
+
+    class Meta:
+        managed = True
+
+    def __str__(self):
+        return '{} File: {}, stored in {} with MD5: {}'.format(self.id, self.name, self.path, self.checksum)
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/models.py]: Please add the newline at the end of file.

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/tests.py [hunk 10]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/tests.py:12]: Please add the copyright.

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/urls.py [hunk 11]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/urls.py]: Please add the copyright.

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/urls.py [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+from django.urls import path
+from . import views
+from django.views.decorators.csrf import csrf_exempt
+
+urlpatterns = [
+    path('', csrf_exempt(views.Recognizer.as_view()))
+]
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/urls.py]: Please add the newline at the end of file

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/utils/checksum.py [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+# with Jalasoft.
+#
+import hashlib
+
+
+class Checksum:
+    """"Calculate Checksum hash from a received file"""
+
+    @staticmethod
+    def md5(file):
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/utils/checksum.py:20]: Please add the description of the method

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/utils/unzip.py [hunk 14]
```diff
@@ -9,11 +9,15 @@
...
 
-from django.core.asgi import get_asgi_application
+class Unzip:
+    """Unzip file on same zip-file path"""
 
-os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'machine_learning.settings')
-
-application = get_asgi_application()
+    @staticmethod
+    def extract(path, filename):
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/utils/unzip.py:20]: Please add the description of the method

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/views.py [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+# with Jalasoft.
+#
+
+from django.views import View
+from django.http import JsonResponse
+from pathlib import Path
+from .model.prediction import Prediction
+from .utils.checker import Checker
+from .utils.unzip import Unzip
+# from .utils.image_recognizer import Image_recognizer
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/views.py]: Please check this import.

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/views.py [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+
+
+class Recognizer(View):
+    """ Machine Learning Endpoint, call machine learning modules with
+        received parameters and recognize objects from a zipped image folder"""
+    def post(self, request):
```
Comment (Franco98404) [inline MACHINE_LEARNING/image_recognizer/image_recognizer_app/views.py:24]: Please add the description of the method

**Review by Franco98404 — COMMENTED** [review]
Please check the comments.

File: MACHINE_LEARNING/image_recognizer/image_recognizer/asgi.py (hunk 1 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer/asgi.py:12]: ok, thanks.

File: MACHINE_LEARNING/image_recognizer/image_recognizer/urls.py (hunk 2 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer/urls.py:12]: done!

File: MACHINE_LEARNING/image_recognizer/image_recognizer/wsgi.py (hunk 3 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer/wsgi.py:12]: done

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/admin.py (hunk 4 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer_app/admin.py:12]: done

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/apps.py (hunk 5 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer_app/apps.py:12]: done

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/exceptions/file_exception.py (hunk 6 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer_app/exceptions/file_exception.py:12]: done

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/exceptions/zip_exception.py (hunk 7 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer_app/exceptions/zip_exception.py:12]: done!

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/model/prediction.py (hunk 8 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer_app/model/prediction.py]: ok

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/tests.py (hunk 10 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer_app/tests.py:12]: ok

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/urls.py (hunk 12 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer_app/urls.py]: done

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/utils/checksum.py (hunk 13 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer_app/utils/checksum.py:20]: ok

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/utils/unzip.py (hunk 14 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer_app/utils/unzip.py:20]: ok

File: MACHINE_LEARNING/image_recognizer/image_recognizer_app/views.py (hunk 15 above)
Comment (lcarrietas) [reply MACHINE_LEARNING/image_recognizer/image_recognizer_app/views.py]: Ok, it was for a new class that finally I wont use.

**Review by Franco98404 — APPROVED** [review]
Now it's ok!

**Review by awielern — APPROVED** [review]
Good job! It looks ok

**Review by MatiasBonifacio — APPROVED** [review]
Everything looks good, i've corrected some minor spelling errors. Good job!

**Pull request merged**